Project.toml

```diff
@@ -19,6 +18,7 @@ ADTypesEnzymeCoreExt = "EnzymeCore"
 
 [compat]
 ChainRulesCore = "1.0.2"
+Compat = "3.47, 4.10"
```

Comment:
are we okay with this additional dependency @ChrisRackauckas?

Reply:
https://discourse.julialang.org/t/what-does-the-new-public-keyword-actually-do/108975/12?u=gdalle

Reply:
I can also copy-and-paste the relevant code from Compat.jl if you want to avoid the 4 stdlib deps Compat.jl has: https://github.com/JuliaLang/Compat.jl/blob/8819abe252579b610cb48c5ad83d51b45a90ddba/Project.toml#L5-L9

Reply:
Or the shorter version
```julia
VERSION >= v"1.11.0-DEV.469" && eval(Meta.parse("public foo, bar"))
```

Straight from the horse's mouth: https://discourse.julialang.org/t/is-compat-jl-worth-it-for-the-public-keyword/119041/2?u=gdalle

Reply:
Oh god... 😅

Reply:
Removed the dependency on Compat in aae032a.

Reply:
I tried 
```Julia
if VERSION >= v"1.11.0-DEV.469"
    eval(Meta.parse("begin
        # Automatic Differentiation
        public AbstractMode
        public ForwardMode, ReverseMode, ForwardOrReverseMode, SymbolicMode
        public mode

        # Sparse Automatic Differentiation
        public Auto
        public dense_ad

        # Sparsity detection
        public sparsity_detector
        public NoSparsityDetector
        public KnownJacobianSparsityDetector
        public KnownHessianSparsityDetector

        # Matrix coloring
        public coloring_algorithm    
        public NoColoringAlgorithm
    end"))
end
```
but the parser seems to have trouble with this, throwing a `ParseError`:
```
ERROR: ParseError:
# Error @ REPL[27]:20:7
    public NoColoringAlgorithm
#     ┌─────────────────────────
end"))    eval(Meta.parse("begin
    # Automatic Differentiation
⋮
    public NoColoringAlgorithm
    end"))
#────────┘ ── extra tokens after end of expression
Stacktrace:
 [1] top-level scope
   @ none:1
```

Reply:
https://discourse.julialang.org/t/is-compat-jl-worth-it-for-the-public-keyword/119041/5

Reply:
Let's wait a bit until the discussion on discourse has settled.

Reply:
The macro-based approach is much more readable than `eval(Meta.parse(...))`.